File: string/check-whether-two-strings-are-almost-equivalent/check-whether-two-strings-are-almost-equivalent.go
package checkwhethertwostringsarealmostequivalent

func checkAlmostEquivalent(word1 string, word2 string) bool {
	count := [26]rune{}

	for _, c := range word1 {
		count[c-'a']++
	}
	for _, c := range word2 {
		count[c-'a']--
	}

	for _, el := range count {
		if el > 3 || el < -3 {
			return false
		}
	}

	return true
}

File: string/check-whether-two-strings-are-almost-equivalent/check-whether-two-strings-are-almost-equivalent.ts
function checkAlmostEquivalent(word1: string, word2: string): boolean {
    const frequency = new Array<number>(26).fill(0);

    for (let c of word1) {
        frequency[c.charCodeAt(0) - 'a'.charCodeAt(0)]++
    }
    for (let c of word2) {
        frequency[c.charCodeAt(0) - 'a'.charCodeAt(0)]--
    }

    for (let f of frequency) {
        if (f > 3 || f < -3) {
            return false;
        }
    }

    return true;
};
